tests/transforms/test_defer_measurements.py

```diff
@@ -1005,8 +1061,37 @@ def qnode2(parameters):
             assert np.allclose(op1.data, op2.data)
 
 
-class TestQubitReset:
-    """Tests for the qubit reset functionality of `qml.measure`."""
+class TestQubitReuseAndReset:
+    """Tests for the qubit reuse/reset functionality of `qml.measure`."""
+
+    def test_new_wire_for_multiple_measurements(self):
+        """Test that a new wire is added if there are multiple mid-circuit measurements
+        on the same wire."""
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def circ(x, y):
+            qml.RX(x, 0)
+            qml.measure(0)
+            qml.RY(y, 1)
+            qml.measure(0)
+            qml.RZ(x + y, 1)
+            qml.measure(0)
+            return qml.expval(qml.PauliZ(1))
+
+        _ = circ(1.0, 2.0)
+
+        expected = [
+            qml.RX(1.0, 0),
+            qml.CNOT([0, 2]),
+            qml.RY(2.0, 1),
+            qml.CNOT([0, 3]),
+            qml.RZ(3.0, 1),
+        ]
+
+        assert len(circ.qtape.operations) == 5
+        for op, exp in zip(circ.qtape.operations, expected):
+            assert qml.equal(op, exp)
```

Comment:
😎 
```suggestion
        assert circ.qtape.operations == expected
```
side-note: I find it weird that `circ.tape` is updated. maybe it's just because batch transforms couldn't possibly update `QNode.tape` (since they might create many tapes) but it's not like we guarantee that `QNode.tape` is the tape we actually execute. anyway ignore me, just rambling

Reply:
It's because qnode construction explicitly applies `defer_measurements` rather than queuing it in the transform program. There will be a PR in the future to clean up transforms related changes. I'll update this there.